libraries/AP_Baro/AP_Baro_UAVCAN.cpp

```diff
@@ -37,6 +37,31 @@ AP_Baro_UAVCAN::~AP_Baro_UAVCAN()
     }
 }
 
+AP_Baro_Backend *AP_Baro_UAVCAN::probe(AP_Baro &baro)
+{
+    AP_Baro_UAVCAN *sensor = nullptr;
+
+    if (AP_BoardConfig_CAN::get_can_num_ifaces() != 0) {
+        for (uint8_t i = 0; i < MAX_NUMBER_OF_CAN_DRIVERS; i++) {
+            if (hal.can_mgr[i] != nullptr) {
+                AP_UAVCAN *uavcan = hal.can_mgr[i]->get_UAVCAN();
+                if (uavcan != nullptr) {
+                    uint8_t freebaro = uavcan->find_smallest_free_baro_node();
+                    if (freebaro != UINT8_MAX) {
+                        sensor = new AP_Baro_UAVCAN(baro);
+                        if (!sensor->register_uavcan_baro(i, freebaro)) {
```

Comment:
You need to do a break or return the sensor when this is successful, otherwise you can lose instances.